src/mjolnir/elevationbuilder.cc

```diff
@@ -64,7 +65,7 @@ void add_elevation(const boost::property_tree::ptree& pt,
     lock.unlock();
 
     // Get the tile. Serialize the entire tile?
-    GraphTileBuilder tilebuilder(graphreader.tile_dir(), tile_id, true);
+    GraphTileBuilder tilebuilder(graphreader.tile_dir(), tile_id, true, include_osmids);
```

Comment:
in all the cases like these where we are reloading an already existing tile, we shouldnt need to add parameters to tell if the ids are included. the graphtileheader that was written with the tile that we are reloading already knows that so we can remove this unneeded change. i think there are several more. the only place where we will need it is likely in `graphbuilder.cc` where we make a new blank one. though maybe in the hierarchy builder it wont be so straight forward. i didnt check it

Reply:
Yeah, the only trick is that what if I want to build tiles that do not contain OSM ids 'on top of' tiles that do? Check the logic in the implementation of the constructor to see what I mean.

Yes... this is in *a lot* of places.

Reply:
i dont understand what you mean by "build on top of" existing tiles. we have two modes of graphtilebuilder essentially, a fresh one with nothing in it and one that initialized from an existing tile. in the former, if you add stuff to the node ids vector then we assume we will be storing node ids (unless, as you put checks in the code, someone doesnt put the right number of these to match the number of graph nodes). in the latter, i we just do whatever we see in the tile, if it had ids we deserialize them and if it didnt we have an empty vector of ids (and the header should be set to false).

i dont think there is a case where you start without osm node ids in the tile and then add them later. at the very least we dont support that as it can only happen between the parsing phases of the build and the graphbuilder phase of the build where we have those `OSMNode` still present on the disk to copy the ids out of.

i looked at the constructor but it didnt dawn on me what you were trying to point me at

Reply:
It is probably me that is not understanding the building process. What I *think* is going on is that we build a set of tiles at some resolution, the refine (or is it coarsen?) them. All my point is, and I don't know why one might want to do it, is that someone might want to include OSM ids at the 'initial' resolution but not at the others, so they would not copy the OSM ids as they copy the nodes.

In the constructor of GraphTileBuilder, the builder only gets to have `has_osmids_` set to true if `include_osmids` is true *and* the tile from which it is initialised has OSM ids (probably should have just said that rather than saying 'go look at...', sorry).

This was just a 'what if' idea that reflects my inexperience and is clearly not a useful option. I think your proposition of inferring intent by just setting it ids is much cleaner.

Reply:
while it is possible to do what you are saying, its quite strange and definitely not a normal case. so our whole tile build is configuration driven. the tile build does have phases and you can stop at one phase (process terminates gracefully) and then restart it at the next phase and continue. to do what you are saying one would have to run the tile building with the ids turned on. wait for it to stop. edit their config to turn it off, and continue the build. in practice i think there is very little use for it but you are right that it is a possible "feature". i think for the sanity of us though it'd be better off if we stick to what was in the config at the beginning bake it into the header from the start and keep on rolling with it.

Reply:
Agreed... it is all gone now and *much* cleaner.

Reply:
Constructor now untouched (simply adding OSM ids signals their presence).